card/mod/machines/lib/javascript/decko_navbox.js.coffee

```diff
@@ -1,72 +1,69 @@
 $(window).ready ->
-#navbox mod
-  $('._navbox').autocomplete {
-    html: 'html',
-    source: navbox_results,
-    select: navbox_select
-# autoFocus: true,
-# this makes it so the first option ("search") is pre-selected.
-# sadly, it also causes odd navbox behavior, resetting the search term
-  }
+  navbox = $('._navbox')
+  navbox.select2
+    placeholder: navbox.attr("placeholder")
+    escapeMarkup: (markup) ->
+      markup
+    minimumInputLength: 1
+    maximumSelectionSize: 1
+    ajax:
+      url: decko.path ':search.json'
+      data: (params) ->
+        query: { keyword: params.term }
+        view: "complete"
+      processResults: (data) ->
+        results: navboxize(data)
+      cache: true
+    templateResult: formatNavboxItem
+    templateSelection: formatNavboxSelectedItem
 
-#navbox mod
-reqIndex = 0 #prevents race conditions
+  navbox.on "select2:select", (e) ->
+    navboxSelect(e)
 
-navbox_results = (request, response) ->
-  f = this.element.closest 'form'
-  formData = f.serialize() + '&view=complete'
+formatNavboxItem = (i) ->
+  if i.loading
+    return i.text
+  '<i class="material-icons">' + i.icon + '</i>' +
+  '<span class="navbox-item-label">' + i.prefix + ':</span> ' +
+  '<span class="navbox-item-value">' + i.label + '</span>'
 
-  this.xhr = $.ajax {
-    url: decko.path ':search.json'
-    data: formData
-    dataType: "json"
-    wagReq: ++reqIndex
-    success: ( data, status ) ->
-      response navboxize(request.term, data) if this.wagReq == reqIndex
-    error: ->
-      response [] if this.wagReq == reqIndex
-  }
+formatNavboxSelectedItem = (i) ->
+  unless i.icon
+    return i.text
+  '<i class="material-icons">' + i.icon + '</i>' +
+  '<span class="navbox-item-value">' + i.label + '</span>'
 
-navboxize = (term, results) ->
+navboxize = (results) ->
   items = []
+  term = results.term
+  if results["search"]
+    # id is what the form sends
+    items.push navboxItem(prefix: "search", id: term, text: term)
 
-  $.each ['search', 'add', 'new'], (index, key) ->
+  $.each ['add', 'new'], (index, key) ->
     if val = results[key]
-      i = {
-        value: term,
-        prefix: key,
-        icon: 'add',
-        label: '<strong class="highlight">' + term + '</strong>'
-      }
-      if key == 'search'
-        i.icon = key
-        i.term = term
-      else if key == 'add'
-        i.href = 'card/new?card[name]=' + encodeURIComponent(val)
-      else if key == 'new'
-        i.type = 'add' # for icon
-        i.href = 'new/' + val[1]
```

Comment:
I wasn't aware of this feature, that you can also create cards of a type that you enter in the navbox.